job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/JobsService.java

```diff
@@ -195,14 +213,20 @@ public Completable deleteJob(UUID jobId, rx.Scheduler scheduler) {
     }
 
     public Observable<ResultSet> insert(Date timeSlice, JobDetails job) {
+        if (job.getJobId() == null || job.getJobName() == null || job.getJobType() == null || job.getTrigger() == null) {
+            logger.warn("Tried to insert job on scheduled jobs with invalid values");
+        }
         return session.execute(insertScheduled.bind(timeSlice, job.getJobId(), job.getJobType(), job.getJobName(),
                 job.getParameters().getMap(), getTriggerValue(session, job.getTrigger())));
     }
 
     public Observable<ResultSet> updateStatusToFinished(Date timeSlice, UUID jobId) {
-        return session.execute(updateStatus.bind((byte) 1, timeSlice, jobId))
+        // First check if the job exists
+        return session.executeAndFetch(findByIdAndSlice.bind(timeSlice, jobId)).flatMap(
```

Comment:
I would consider adding some logging in here. The expectation is that there will be a row for `timeSlice` and `jobId`. At a minimum, I would log a warning that the row does not exist. Then the question is, what if anything else should be done?

Reply:
I think for now just logging and then ignore it and of course don't try to do the upsert operation.